### DISCUSSION TITLE
Does adding a post, or editing an existing one require rebuilding the project?

### DISCUSSION THREAD

**Discussion by ponteshu** [dbernheisel/bernheisel.com#18] [General]
The posts are in `/posts/**.md`, and they're statically retrieved into a variable -- `@posts`, or module attribute.

Does this mean that adding a post, or editing an existing one requires rebuilding and redeploying the project?

**Comment by dbernheisel** [comment]
That is correct, all posts are compiled and in memory. Updating a post requires recompiling that module.

**Comment by ponteshu** [comment]
This saves the posts into a module attribute which then requires recompilation of the project, and each time time there's a change:
```
  use NimblePublisher,
    build: Bern.Blog.Post,
    from: "posts/**/*.md",
    as: :posts,
    highlighters: [:makeup_elixir, :makeup_diff]
```

Do you know if there'll any **simple way** to make retrieving posts dynamically instead?
In order to avoid recompilation.
Meaning, the posts may remain in the same .md files, but I want to avoid creating a module attribute in which they're get stored and built into the code itself.

**Reply by dbernheisel** [reply]
Sure, you might consider using a markdown rendering engine compatible with Phoenix. This runs during runtime so it's less efficient (comparatively to compile-time) but does not require the project to be recompiled.

https://github.com/boydm/phoenix_markdown

I have not used it so I cannot attest to it working well or being simple, but if it does as it says, looks good to me.

**Reply by dbernheisel** [reply]
On second thought, that phoenix_markdown library may also require recompilation, I'm not sure, since it's also eex under the hood which compiles templates except for the dynamic bits which will evaluate at runtime. 

In other words, you'd have to consider the whole markdown file as the dynamic bit and read it from file on each request (or some cache system to invalidate based on file stats). 

You might consider https://stackoverflow.com/questions/13223238/how-do-you-create-and-load-modules-dynamically-at-runtime-in-elixir-or-erlang but honestly I'm not sure it's a good idea. Many container-based deployments run ephemerally with read-only volumes, so recompiling dynamically might not work since it may write the resulting beams to file. You'd need a read/write volume. All this I'm not 100% sure on so it just requires some experimentation.
